storage/src/backend/registry.rs

```diff
@@ -188,6 +189,10 @@ struct RegistryState {
     // Use RwLock here to avoid using mut backend trait object.
     // Example: RwLock<"Bearer <token>">
     //          RwLock<"Basic base64(<username:password>)">
+
+    // Cache the method of getting auth, get | post
+    cached_get_auth_method: HashCache,
```

Comment:
How about:

``` rust
// Cache for the HTTP method when getting auth, possible values: "get", "post".
// Due to the different implementations of various image registries, auth requests
// may use the GET or POST methods, we need to cache the method after the
// fallback, so it can be reused next time and reduce an unnecessary request.
cached_auth_using_http_get: HashCache,
```

And can we make it as `HashCache<bool>` ?

```
struct HashCache(RwLock<HashMap<String, T>>);
```

Reply:
In the context ,HashCache is implemented as 
`struct HashCache(RwLock<HashMap<String, String>>);`
if  we make it as' HashCache<bool> ', it need implement another logic

Reply:
I mean the generic type https://doc.rust-lang.org/rust-by-example/generics.html

Reply:
I have resolve it with changing the implemention of `struct HashCache(RwLock<HashMap<String, String>>);` to `struct HashCache<T>(RwLock<HashMap<String, T>>);`.